src/components/scenes/ChangePasswordScene.tsx

```diff
@@ -79,48 +83,49 @@ const ChangePasswordSceneComponent = ({
     .some(([, value]) => value === 'unmet' || value === 'error')
 
   const handleNext = useHandler(async () => {
-    if (password !== confirmPassword) {
-      setConfirmPassword('')
-      setPasswordReqs({
-        ...passwordReqs,
-        confirmationMatches: 'error'
-      })
-      return
-    }
-
     setSpinning(true)
 
-    try {
-      await onSubmit(password)
-    } catch (e) {
-      showError(e)
-      setSpinning(false)
+    handleValidatePassword(password, 'error')
+    if (
+      !Object.entries(passwordReqs).some(
+        ([, value]) => value === 'unmet' || value === 'error'
+      )
+    ) {
+      try {
+        await onSubmit(password)
+      } catch (e) {
+        showError(e)
+      }
     }
+
+    setSpinning(false)
   })
 
   const handleFocusSwitch = () => {
     setFocusFirst(false)
     setFocusSecond(true)
   }
 
-  const handleValidatePassword = (password: string) => {
-    setPassword(password)
-
-    setPasswordReqs({
-      minLengthMet: password.length >= 10 ? 'met' : 'unmet',
-      hasNumber: /[0-9]/.test(password) ? 'met' : 'unmet',
-      hasLowercase: /[a-z]/.test(password) ? 'met' : 'unmet',
-      hasUppercase: /[A-Z]/.test(password) ? 'met' : 'unmet'
-    })
-  }
+  const handleValidatePassword = useHandler(
```

Comment:
This should not be a handler, but rather a utility method, since it's not passed to props of any component. So I would call it `validatePassword`, not `handleValidatePassword`